src/nanoarrow/hpp/view.hpp

```diff
@@ -157,6 +158,7 @@ class ViewArrayAs {
                 array_view->buffer_views[1].data.data,
                 array_view->offset,
             },
+            array_view->offset,
```

Comment:
This is an easy fix but maybe not the API we want, since we are providing the offset to both the Range and the `Get` class template (?)